solver/types.go

```diff
@@ -9,29 +9,43 @@ import (
 	ocispec "github.com/opencontainers/image-spec/specs-go/v1"
 )
 
-// Vertex is one node in the build graph
+// Vertex is a node in a build graph. Vertex defines an interface for a
```

Comment:
nit: Starting both sentences with "Vertex" feels redundant. Maybe "Vertex is a node in a build graph. It defines an interface…"